engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Builtins.java

```diff
@@ -301,4 +309,48 @@ public ModuleScope getScope() {
   public Module getModule() {
     return module;
   }
+
+  /**
+   * Convert from type-system type names to atoms.
+   *
+   * @param typeName the fully qualified type name as defined in {@link Constants}.
+   * @return the associated {@link org.enso.interpreter.runtime.callable.atom.Atom} if it exists,
+   *     and {@code null} otherwise
+   */
+  public Atom fromTypeSystem(String typeName) {
+    switch (typeName) {
+      case Constants.ANY:
+        return any.newInstance();
+      case Constants.ARRAY:
+        return mutable.array().newInstance();
+      case Constants.BOOLEAN:
+        return bool.getBool().newInstance();
+      case Constants.DECIMAL:
+        return number.getDecimal().newInstance();
+      case Constants.ERROR:
+        return dataflowError.constructor().newInstance();
+      case Constants.FUNCTION:
+        return function.newInstance();
+      case Constants.INTEGER:
+        return number.getInteger().newInstance();
+      case Constants.MANAGED_RESOURCE:
+        return resource.getManagedResource().newInstance();
+      case Constants.NOTHING:
+        return nothing.newInstance();
+      case Constants.NUMBER:
+        return number.getNumber().newInstance();
+      case Constants.PANIC:
+        return panic.newInstance();
+      case Constants.REF:
+        return mutable.ref().newInstance();
+      case Constants.TEXT:
+        return text.getText().newInstance();
+      case Constants.THUNK:
```

Comment:
This should not exist. It is incorrect to expect a thunk, ever. The runtime may decide to pass an evaluated value in the place of a thunk, without notice.